src/Microsoft.AspNet.Server.Kestrel/Infrastructure/MemoryPoolIterator2.cs

```diff
@@ -137,6 +137,28 @@ public int Peek()
             }
         }
 
+        public MemoryPoolIterator2 Skip(int length)
+        {
+            var block = _block;
+            var index = _index;
+            var remaining = length;
+            var following = block.End - index;
+            while (true)
+            {
+                if (remaining <= following)
+                {
+                    return new MemoryPoolIterator2(block, index + remaining);
+                }
+                if (block.Next == null)
+                {
+                    return new MemoryPoolIterator2(block, index + following);
+                }
+                remaining -= following;
+                block = block.Next;
+                index = block.Start;
```

Comment:
The three branches of condition will be more readable if organized in the form of `if - else if - else`